code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -183,7 +186,8 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 			for(var/obj/item/W in summoner)
 				if(!summoner.dropItemToGround(W))
 					qdel(W)
-			summoner.dust()
+			if(dusts_master)
+				summoner.dust()
```

Comment:
Might also wanna put the message for the dusting aswell as the part dropping all their items behind the dusts_master check.